labshare/tests.py

```diff
@@ -1551,13 +1745,61 @@ def test_overview_done_button(self):
         num_reservations_for_user = Reservation.objects.filter(user=self.staff_user).count()
         gpu = self.driver.find_element_by_id(self.device_1.gpus.first().uuid)
         done_button = gpu.find_element_by_class_name("gpu-done-button")
-        self.assertNotIn("hidden", done_button.get_attribute("class"))
+        self.assertNotIn("d-none", done_button.get_attribute("class"))
         done_button.click()
 
         self.wait_for_page_load()
         self.assertEqual(Reservation.objects.filter(user=self.staff_user).count(), num_reservations_for_user - 1)
 
 
+@skipIf("TRAVIS" in os.environ and os.environ["TRAVIS"] == "true", "Skipping this test on Travis CI.")
+class FrontendOverviewExtendButtonTest(FrontendTestsBase):
+
+    def test_extend_button(self):
+        self.wait_for_page_load()
+
+        gpu = self.driver.find_element_by_id(self.device_1.gpus.last().uuid)
+        extend_button_group = gpu.find_element_by_class_name("gpu-extend-button-group")
+        self.assertNotIn("d-none", extend_button_group.get_attribute("class"))
+        extend_button = extend_button_group.find_element_by_class_name("gpu-extend-button")
+        extend_button.click()
+
+        self.wait_for_page_load()
+        reservation = Reservation.objects.filter(gpu=self.device_1.gpus.last()).first()
+        self.assertGreater(abs(utc_now() - reservation.usage_expires), Reservation.usage_period() - timedelta(minutes=1))
+
+        gpu = self.driver.find_element_by_id(self.device_1.gpus.last().uuid)
+        extend_button_group = gpu.find_element_by_class_name("gpu-extend-button-group")
+        self.assertIn("d-none", extend_button_group.get_attribute("class"))
+
+        done_button = gpu.find_element_by_class_name("gpu-done-button")
+        self.assertNotIn("d-none", done_button.get_attribute("class"))
+
+
+@skipIf("TRAVIS" in os.environ and os.environ["TRAVIS"] == "true", "Skipping this test on Travis CI.")
+class FrontendOverviewDropdownButtonTest(FrontendTestsBase):
+
+    def test_dropdown_done_button(self):
+        self.wait_for_page_load()
+
+        self.driver.implicitly_wait(1)
+
+        num_reservations_for_user = Reservation.objects.filter(user=self.staff_user).count()
+        gpu = self.driver.find_element_by_id(self.device_1.gpus.last().uuid)
+
+        extend_button_group = gpu.find_element_by_class_name("gpu-extend-button-group")
+        self.assertNotIn("d-none", extend_button_group.get_attribute("class"))
+
+        dropdown_toggle_button = extend_button_group.find_element_by_class_name("dropdown-toggle")
+        dropdown_toggle_button.click()
+
+        done_entry = extend_button_group.find_element_by_class_name("dropdown-item")
+        done_entry.click()
+
+        self.wait_for_page_load()
+        self.assertEqual(Reservation.objects.filter(user=self.staff_user).count(), num_reservations_for_user - 1)
+
```

Comment:
Could you also check that the `reserve` button is visible from now on?

Reply:
If we want to test that, we should probably do it, in the actualy done button test too.

Reply:
that sounds reasonable^^